addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -416,6 +416,53 @@ def _rtc_invite_members(self, member_ids=None):
             self.env['bus.bus']._sendone(self.channel_id, 'mail.record/insert', {'Thread': channel_data})
         return members
 
+    def _mark_as_read(self, last_message_id, sync=False):
+        """
+        Mark channel as read by updating the seen message id of the current
+        member as well as its new message separator.
+
+        :param last_message_id: the id of the message to be marked as read.
+        :param sync: wether the new message separator and the unread counter in
+            the UX will sync to their server values.
+        """
+        self.ensure_one()
+        domain = [
+            ("model", "=", "discuss.channel"),
+            ("res_id", "=", self.channel_id.id),
+            ("id", "<=", last_message_id),
+        ]
+        last_message = self.env['mail.message'].search(domain, order="id DESC", limit=1)
+        if not last_message:
+            return
+        self._set_last_seen_message(last_message)
+        self._set_new_message_separator(last_message.id + 1, sync=sync)
+
+    def _set_last_seen_message(self, message, notify=True):
```

Comment:
Inconsistent param name/type between methods (message is given once by id, once by record), and sync/notify looks like the same idea too.

For message param, I'd suggest recordset instead of id, as we keep records as much as possible in internal code.